### ISSUE TITLE
Admin action / co-located ecosystem

### ISSUE THREAD

**Issue by laurjpeterson** [department-of-veterans-affairs/caseflow#5434]
This epic encompasses the creation of administrative actions by attorneys, their transfer to co-located, and their resolution by co-located support. 

- Attorney to co-located
- Co-located auto-case distribution
- Co-located queue

See epic [mural](https://app.mural.co/t/workqueue2001/m/workqueue2001/1524169387338/8917c31aab5f57444eaa6aacf5b93bc310395e29) for design progression.

---

# Attorney to Co-located

## Background
Co-located admin support complete administrative actions assigned to them by attorneys. Administrative actions include things like hearing clarifications, POA clarifications, IHP requests, etc

Attorneys currently assign cases to co-located using VACOLS, assigning cases to Team 1 / location 11 or Team 2 / location 12. Then, they send an email to a shared inbox with instructions for the administrative action. 

## Goals
- Allow attorneys to assign admin actions to co-located admin in Queue instead of VACOLS
- Eliminate the use of email for admin actions
- Explore attorney ability to complete certain admin actions where possible - e.g. could attorneys place cases on hold for extension requests? could attorneys assign cases directly to translation?
- Give attorneys visibility into cases that are at co-located/in abeyance and the ability to follow-up more easily
- Ultimately, unblock attorneys from writing decisions as quickly as possible

### Attorneys assigning admin actions to co-located
- When an attorney is reviewing a Veteran's claims folder and identifies that an admin action needs to be done, they need a way to assign it to co-located support, so the admin action can be completed while the attorney works on another case. (from Case details) - https://github.com/department-of-veterans-affairs/caseflow/issues/6031

- **Multiple admin actions: Allow an attorney to assign multiple admin actions to a co-located user** - https://github.com/department-of-veterans-affairs/caseflow/issues/5936 _Update 07/23/18: We're postponing this feature (from a front-end perspective) for now_

- When an attorney is assigning an admin action, they need to choose the type of admin action they are assigning, so co-located knows what their task is. #5439 

- When an attorney is assigning an admin action, they need to explain instructions for the task, so co-located knows how to execute on that task. #5439 

- When an attorney assigns an admin action to co-located, the case should be moved to the attorney's on hold queue, so the attorney can monitor the case while it is on hold.  _Update 07/23/18: We're postponing this feature (from a front-end perspective) for now_

### Attorneys assigning admin actions to Translation
- When an attorney identifies a document in the Veteran's claims folder requiring translation, they want to assign the case to translation directly and include instructions about which document requires translation, so the translation can be done as quickly as possible. _Mocks TBD_

### ~Attorneys completing admin actions themselves~
- holding off on this! 

#### Placing holds / abeyance
- When an attorney identifies the Veteran or POA has requested an extension, they want to place the case on hold directly, so the 30, 60, or 90 day hold can begin as early as possible.
  - **Open question:** Do these cases also need to go to a co-located team member if additional mail comes in for the Veteran? 

- When an attorney places a case on hold, they need to define how long the hold lasts, so they can continue working on it when it is ready. _TBD_

- When an admin action's hold is up, co-located team members want to be alerted, so they can continue working on the case ASAP and move the appeal along in the process. _TBD_

- When an attorney places a case on hold, it should not be counted against the attorney's production goals, because they cannot act on the case until they are unblocked. _TBD_

**Judge-related**
- When an attorney has placed a case on hold, judges should see that the case is on hold in some way, so they do not badger the attorney for holding the case too long. _TBD_
  - Perhaps the counter stops? 

## Open Questions
- If attorneys place cases on hold for extension requests, how does that impact mail routing? If we do not put this case on co-located's plate, would attorneys then be required to address that mail? Is that ok?
- How should Judges view cases attorneys have put on hold because they cannot act on them?
  - Attorney 'days waiting' should pause?
  - Another marker to show the judge the case is currently at co-located/on hold

## Resources

---

# Co-located auto case distribution

## Background
When attorneys assign cases to co-located, they choose team 1/location 11 or team 2/location 12 (aligned with the two DVCs). Previously, individual co-located team members manually assigned these admin actions among the team. Now, they use a batch case assignment tool Jed built into VACOLS. 

## Goals
- Automatically distribute cases requiring administrative action among co-located team members, as evenly as possible
- Reuse learnings from case storage auto-assignment to judges

## Proposed scope
- In scope: Locations 11 and 12 - admin actions
- Out of scope: Locations 31 and 32 - step before dispatch

## Requirements
- Assign cases to co-located team members in a round robin fashion

## Open Questions
- How will this account for co-located team member PTO impacting appeals sitting longer than they should? Should have co-located ask for cases similar to how judges will prompt for cases?

---
# Co-located queue

## Background
Co-located admin support complete administrative actions assigned to them by attorneys. Administrative actions include things like hearing clarifications, POA clarifications, IHP requests, etc. Many admin actions require the appeal to be placed on hold (currently, placed in an abeyance location). Process today:
- Attorneys charge appeals to co-located locations 11 and 12. 
- Team leads allocate cases to specific team members within those locations using Jed's allocation program in VACOLS (this previously was a manual process).
- Team members access instructions for the admin action via emails from attorneys
- Team members complete tasks as part of the admin action. This could involve updating appeal data, sending correspondence to Veterans/POA, placing an appeal in abeyance, charging the appeal to other teams (e.g. FOIA, translation, etc.)
- Team members charge the appeal back to judges/attorneys when tasks are finished. 

Co-located also complete mail tasks when appeals are assigned to attorneys and judges. They also play a role in BVA dispatch processes, which will **not** be covered in this spec. 

One way to bring Queue functionality to co-located is to allow attorneys to assign administrative actions to them in Queue, for them to complete admin actions, place them on hold, and assign them back to attorneys. 

## Pain Points from [user research](https://app.mural.ly/t/workqueue2001/m/workqueue2001/1505361321430/0acc8ea64c141623fdc3f7effd7e397e52629228)
- Tasks arrive a large variety of ways
- Shared inbox is difficult to manage with many staff members
- Some tasks are untrackable (they come in via email or "drive-by's")
- Appeals can get lost when OVLJ are not clear on where to route case after admin action is done
- Admin action instructions can be complex and confusing
- No place for each staff person to keep track of own tasks
- Workloads are uneven bc OVLJ favors individual staff
- OVLJ will charge straight to 24 (abeyance), no one knows appeal is there
- Co-located is overworked; each staff fields requests from up to 80 attorneys
- No system for supervisors to keep track of staff's tasks
- Co-located double checking before Dispatch is redundant
- Manually compiling productivity report daily

### User / business goals
- Consolidate co-located cases in one place - prevent attorney reliance on email so co-located can better manage their work
- Facilitate better understanding of tasks to be completed and their instructions
- Facilitate quicker follow-up when cases have completed their holds
- Allow better tracking of cases in abeyance

### Internal goals
- Reuse judge and attorney queue UI for how co-located view and distinguish cases assigned to them and view case details
- Reuse logic from automatic case assignment to judges where possible
- Determine if we need to allow attorneys to assign cases to co-located first, or if that comes after all co-located has queue functionality

### Table/Queue (establish table #6077 )

**#6078 : New Tab**
- When a co-located team member is planning what to work on, they need to see all tasks assigned to them (mail, VACOLS, 3230s, etc.) in one place, so they can prioritize what to work on.
  - These tasks include appeals assigned to them to work AND appeals that they placed on hold.

- When a co-located team member is planning what to work on, I need to see what type of tasks are assigned to me, so I can prioritize what to work on.
  - e.g. task type column
  - this task type would initially come from the VACOLS field, and in the future come from when attorneys assign the tasks in queue

- **Multiple admin actions: When an attorney assigned multiple admin actions for the same case, I need to see both of those admin actions in my table so I have context before I start working on them.** - https://github.com/department-of-veterans-affairs/caseflow/issues/6014 _Update 07/23/18: We're postponing this feature (from a front-end perspective) for now_

### Case Details
**insert case details Issue w/mock here**
- When a co-located team member decides what to work on, they need to see and understand the instructions given by an attorney, so they know how to act on the case assigned to them.
  - Instructions, type of task in the case details screen   

- When a co-located team member decides what to work on, they need to see details about the case, so they can get context, or check specific data about an appeal to complete an admin action.
  - Appeal summary, details, etc. in case details screen

- Multiple admin actions: As a co-located staff member — when an attorney assigns multiple admin actions for the same case, I need to see both of those admin actions in Case details/Case snapshot so I have context before I start working on them. - https://github.com/department-of-veterans-affairs/caseflow/issues/6015 _Update 07/23/18: We're postponing this feature (from a front-end perspective) for now_

### Admin action - tangential activities
- When the Veteran requests an extension request, co-located team members need to send Veterans and POAs letters before putting a case on hold. 
  - Note: this requires some research, see open questions. I do not think Queue will handle letter generation, but we should work it into our flow. _TBD_

- When required by admin action, co-located team members need to be able to update data about an appeal (on the case details screen), to complete the admin action so the attorney can continue working on the appeal. 
  - Note: we will likely want attorneys to be able to update data about an appeal so this story may be rendered obsolete
  - Note: specific stories for this will be handled by case details editing epic

- When required by admin action, co-located team members need to send Veterans and POAs letters before putting a case on hold. 

### On hold / abeyance 
**#6080 : On Hold Tab (Table)**
** # : Place on Hold flow**
- When an administrative action requires further actions before assigning back to an attorney (e.g. extension request, wait for IHP or POA clarification), co-located team members need to put the case on hold.
  - This does not remove the case from the co-located team member's list of work, but recognizes that further work cannot be done on it at this time. It should not count against the user's productivity standards by still being on the team member's plate.

- When an administrative action requires further actions before assigning back to an attorney (e.g. extension request, wait for IHP or POA clarification), co-located team members need to define how long the hold lasts, so they can continue working on it when it is ready.

### Pending Action Tab (Table)
**#6079 : Pending Action Tab (Table)**
- When an admin action's hold is up, co-located team members want to be alerted, so they can continue working on the case ASAP and move the appeal along in the process.

### Assigning to other teams
**Insert Issue w/mock here**
- When required by admin actions, co-located team members need to assign cases to other teams, so the admin action can be completed by the right team.
  - Likely include: FOIA, OMO, Translation, Hearings
  - Note: we may want attorneys to assign cases directly to translation for certain cases, too.

### Completed admin actions / assigning back to attorneys
**Insert Issue w/mock here**
- When a co-located team member has completed an administrative action, they need to assign the case back to the user (attorney/judge/etc.) who assigned it, so they can continue working on the case.

### Out of Scope for this epic 
- Auto-assignment of tasks to co-located - #716
- Decision related tasks - need decision on whether co-located or BVA dispatch will be doing these checks
- Supervisor views
- Co-located productivity reports

## Open questions
- Do team 1 and team 2 still exist now that there is only one DVC?
- How do co-located team members currently access, print, and send letters to Veterans and POA when required for an admin action? Do they do this in VACOLS?

## Resources
- [January 2018 co-located usability testing](https://github.com/department-of-veterans-affairs/appeals-design-research/tree/master/Projects/BVA%20Work%20Queue/Usability%20Testing/Co-located)
- [Co-located invision prototypes](https://va-gov.invisionapp.com/d/main/#/console/12439412/261192573/preview)
- [Interview with Pam](https://github.com/department-of-veterans-affairs/appeals-design-research/blob/master/Projects/BVA%20Work%20Queue/Post-Discovery/Co-Located/Pam_Co-Located%20Case%20Assignment.md)
- [Co-located Mural](https://app.mural.ly/t/workqueue2001/m/workqueue2001/1505361321430/0acc8ea64c141623fdc3f7effd7e397e52629228)
- [Job stories across teams](https://app.mural.ly/t/workqueue2001/m/workqueue2001/1508879064985/48ca6c52326e988054a1166b937020b314c516d4)
- [Task Analysis Mural](https://app.mural.ly/t/workqueue2001/m/workqueue2001/1506103905938/aefd1e4b82ed8ea26508ee3bd9190b030e1db141)

## Exploration Notes - 4/18
- How should we display on hold? tabs, or a la judges review x cases, assign y cases.

# Rollout Plan
- See queue releases and rollouts [here](https://github.com/department-of-veterans-affairs/appeals-design-research/blob/master/Projects/Caseflow%20Queue/Launch/Queue%20Rollout%20Plans.md#admin-action-pilot)

# Resources
- Co-located and Caseflow [deck](https://docs.google.com/presentation/d/1qARc-bZQ8jtYWdlcvTiIuDYkm2PMsehYohfTz1pjpBg/edit) - 5.30.2018
- Co-located usability testing synthesis [readout deck](https://docs.google.com/presentation/d/1rw7qbOsGwEDVRa2W0w2GNQgaGW5Lk0ZXuxWK2xQPVkA/edit?usp=drive_web&ouid=104130188314644782843)

### List of admin actions in VACOLS case distribution program
- IHP
- Waiver of AOJ letter
- Arneson letter
- Hearing clarification letter
- POA clarification letter
- Unaccredited representative letter
- Retired VLJ letter
- New rep arguments (PT_VSO) letter
- Extension request ruling
- Missing hearing transcript
- Schedule hearing
- FOIA/Privacy Act request
- Pending scanning banner VBMS
- Missing records in file
- ~Substitution determination needed~
- Address verification
- Other

---
# Exploration Notes

### Decisions
- We agree on MVP excluding On hold functionality. "Check out flow" to co-located is in scope. 

### To dos:
- Explore how attorneys might select more than one admin action. Should each action have notes?
- Craft confirmation screen messaging, other copy/content
- Respond to comments on InVision
- Define usability testing for On hold - aim for week of May 1

### Full-featured
- General questions
  - How should we display on hold? tabs, or a la judges review x cases, assign y cases.
  - How do we explain to an attorney what on hold means? How can we ensure they are at ease?
  - Should attorneys be able to take any action on cases on hold in our system?
- Assignment
  - Should we automatically assign to co-located teams? What are the pros and cons of mapping attorneys/judges to team 1 and team 2 in the back end - it is reflecting BVA org structure?
  - Admin actions drop down - should we add a searchable dropdown? Are 15 entries too much for a dropdown?
  - Success message copy
- On hold table
  - Should the attorney on hold table surface a co-located team member's name? Or not? 
    - How often would an attorney reach out to co-located to ask for information? What is the actionable follow-up?
    - If the case is always assigned to team 1, should attorneys be contacting the team lead?
  - On hold table
    - What instructional text might we add to describe what On hold means?
    - Task type added for on hold
    - Do we allow any actions for attorneys on the On hold table? e.g. cancel?
    - Days on hold vs. date assigned? 
    - How should we calculate days on hold?
       - simplest = today's date - date attorney assigned it off their plate
       - e.g. when admin initiates an action that puts the case on hold? 
- Downstream effects
  - **How do on hold cases affect judges seeing attorneys' queues?**

### MVP
- Craft confirmation screen messaging
- Explore how attorneys might select more than one admin action.

### General questions
- Do attorneys add the admin actions in VACOLS today because of the automatic case distribution program?
- How often do attorneys assign more than one admin action for an appeal? Should these admin actions be split among co-located staff?
- Are we set on the term "admin action"?
- How do we set up the Board employees to adopt the concept of on hold not counting against them?

**Comment by laurjpeterson** [comment]
Meeting with DVC Osborne about co-located requirements - https://github.com/department-of-veterans-affairs/appeals-design-research/issues/857

**Comment by laurjpeterson** [comment]
Diary codes - https://github.com/department-of-veterans-affairs/dsva-vacols/issues/12

**Comment by laurjpeterson** [comment]
Closing - stored in the wiki, here. https://github.com/department-of-veterans-affairs/caseflow/wiki/VLJ-Support
